src/cpp/database/samples.hpp

```diff
@@ -151,6 +161,35 @@ struct ByteCountSample : StatisticsSample
         return !(*this == other);
     }
 
+
+    /**
+     * ByteCountSample is, in a way, a number expressed in base 2^64, where count is the first digit from the right
+     * (LSD), and magnitude_order is the second digit from the right (MSD). However, it has 2 peculiarities:
+     *     1. MSD range is [-2^15; 2^15-1]
+     *     2. LSD range is [0; 2^64-1]
+     *
+     * Because LSD CANNOT be negative, a negative number represented as ByteCountSample is a bit counter intuitive
+     * at first. Following, and example of the representation of the same number in base 10 and as a ByteCountSample
+     * is given (assume that a number can expressed as (MSD, LSD) in both representations).
+     *     BASE 10:
+     *         -9 = (0, -9)
+     *     ByteCountSample
+     *         -9 = (-1, 2^64 - 9) -> This is interpreted as (-(1 * 2^64) + (2^64 - 9)) = -2^64 + 2^64 -9 = -9
+     */
+    inline ByteCountSample operator -(
+            const ByteCountSample& other) const noexcept
+    {
+        ByteCountSample ret(kind);
```

Comment:
>  The operation cannot return a negative number (negative magnitude order).

Not in the specific use tat we are doing of it. However, since a `ByteCountSample` (MSD, LSD) can represent a negative number, then it makes sense that the subtraction operation allows for the minuend to be smaller that the subtrahend (which will yield a negative number). I tried to differentiate between the number representation itself, and the use we do of it on upper layers. The fact that the operation will never be performed in upper layers should not mean that it is not defined.

It is different from the other case in the sense that for `EntityCountSample` there are no negative numbers, so performing the operation where minuend < subtrahend is not defined. "By chance" c++ deals with this case by simply rolling to the unit to the other side, by if we do that, then the resulting `EntityCountSample` would be meaningless.

One check I'm not performing though is when the result would be smaller than the minimum `ByteCountSample` possible (-2^15; 2^64-1). I assumed it'll never happen, but that break my prior argument so I'd be OK with adding it. What do you think?

Reply:
I think you are going to add that test independently of what I think. But yes, I think it would be better to have tests that takes all cases into account.